onnxruntime/core/providers/cuda/tensor/trilu.cc

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/tensor/trilu.h"
+#include "core/providers/cuda/tensor/trilu_impl.h"
+#include "core/providers/cpu/tensor/utils.h"
+
+using namespace onnxruntime::common;
+
+namespace onnxruntime {
+namespace cuda {
+
+ONNX_OPERATOR_KERNEL_EX(
+    Trilu,
+    kOnnxDomain,
+    14,
+    kCudaExecutionProvider,
+    (*KernelDefBuilder::Create())
+        .InputMemoryType(OrtMemTypeCPUInput, 1)
+        .TypeConstraint("T", DataTypeImpl::AllFixedSizeTensorTypes()),
+    Trilu);
+
+Status Trilu::ComputeInternal(OpKernelContext* ctx) const {
+  const Tensor* input_ptr = ctx->Input<Tensor>(0);
+  const auto* k = ctx->Input<Tensor>(1);
+
+  int64_t k_val = 0;
+  if (k) {
+    ORT_ENFORCE(IsScalarOr1ElementVector(k), "k should be a 1-D or 0-D tensor.");
+    k_val = *(k->template Data<int64_t>());
+  }
+  if (input_ptr == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
+  const Tensor& input = *input_ptr;
+  const TensorShape& shape = input.Shape();
+  const std::vector<int64_t>& input_dims = shape.GetDims();
+  int32_t rank = gsl::narrow_cast<int32_t>(input_dims.size());
+  if (rank < 2) {
+    return Status(ONNXRUNTIME, INVALID_ARGUMENT, "Input tensor should have a rank of at least 2");
+  }
+  Tensor* output = ctx->Output(0, shape);
+  auto batch_size = input_dims[rank - 1] * input_dims[rank - 2];
```

Comment:
'batch_size' is usually a name reserved for the leading dimensions. 
Consider a different name.

Reply:
Okay. I will try to find another name. However pl look at 
["Trilu takes one input tensor of shape [*, N, M], where * is zero or more batch dimensions"](https://github.com/onnx/onnx/blob/master/docs/Operators.md#:~:text=Trilu%20takes%20one%20input%20tensor%20of%20shape%20%5B*%2C%20N%2C%20M%5D%2C%20where%20*%20is%20zero%20or%20more%20batch%20dimensions)
In this case all dimensions except the last two dimensions are considered batch dimensions.